expand_load.py

```diff
@@ -0,0 +1,135 @@
+    events is specified, the original data is added multiple times with new
+    random time noise each time. Exceptions are raised if n_pixels or n_events
+    are too small. A scipp dataset is returned.
+
+    Arguments
+    ---------
+
+    filename : str
+        Filename of nexus event datafile to be loaded
+
+    n_pixels : int
+        Number of pixel for returned scipp dataset, must be larger than original
+
+    Keyword Arguments
+    -----------------
+
+    n_events : int
+        Number of events for returned scipp dataset, must be sufficiently large
+
+    time_noise_us : float
+        Width of gaussian noise added to time of flight data
+
+    verbose : Boolean
+        Sets verbose mode which prints additional information
+
+    """
+    # Make sure n_pixels and n_events are ints or can at least be converted
+    n_pixels = int(n_pixels)
+    n_events = int(n_events)
+
+    d = sc.Dataset()
+    d["loaded_data"] = load(filename=filename, load_pulse_times=False)
+    if verbose:
+        print("Loaded dataset")
+        print(d)
+
+    original_length = len(np.array(d.coords[Dim.Position].values))
+    original_events = 0
+    for pixel_id in range(original_length):
+        original_events += len(np.array(d["loaded_data"].coords[Dim.Tof].values[pixel_id]))
+
+    if n_pixels < original_length:
+        raise ValueError("n_pixel less than number of pixels in existing datafile")
+
+    expected_events = n_pixels/original_length*original_events
+    if n_events is not None and n_events < expected_events:
+        print(n_pixels, n_events)
+        print(original_length, original_events)
+        raise ValueError("n_events too small, not all pixels will get events")
+
+    # Adding pixels to dataset
+    current_length = original_length
+    positions_var = d.coords[Dim.Position]
+    combined_var = positions_var.copy()
+    while (current_length <= n_pixels - original_length):
+        current_length += original_length
+        combined_var = sc.concatenate(combined_var, positions_var, Dim.Position)
+
+    remaining_length = n_pixels - current_length
+
+    if remaining_length > 0:
+        combined_var = sc.concatenate(combined_var, positions_var[Dim.Position, 0:remaining_length], Dim.Position)
+
+    ds = sc.Dataset(coords={Dim.Position: combined_var})
+
+    if verbose:
+        print("Generated dataset before tof events are added")
+        print(ds)
+
+    # Adding sparse data to dataset
+    tofs = sc.Variable(dims=[Dim.Position, Dim.Tof], shape=[n_pixels, sc.Dimensions.Sparse], unit=sc.units.us)
+    ds["measurement"] = sc.DataArray(coords={Dim.Tof: tofs})
+
+    added_events = 0
+    while n_events is None or added_events < n_events: # keep adding sparse data until enough events reached
+        original_id = 0
+        for pixel_id in range(n_pixels):
+            # For each pixel, grab the events from the original file
+            pos_n_events = len(np.array(d["loaded_data"].coords[Dim.Tof].values[original_id]))
```

Comment:
I recommend working with `DataArray` instead of `Dataset`. Then you can avoid the artificial (and more lengthy) explicit item names such as `"loaded_data"`.

Why did you convert to `np.array` before taking `len`?

Reply:
Will try to use DataArray in the future, but I don't see how I could have done it in that particular instance? Should I have saved  d["loaded_data"] as another variable and worked with that?

I didn't know scipp had len functionality built in, it is never mentioned in the documentation.

Reply:
Something like:
```python
loaded_data = sc.neutron.load(...)
output = sc.DataArray(coords=...)
```

Reply:
I see, thought you meant internally, but the returned result can just be a DataArray. This branch has been updated with this improvement now. Ready to merge?